libzerocash/Address.cpp

```diff
@@ -11,24 +11,35 @@
  * @copyright  MIT license (see LICENSE file)
  *****************************************************************************/
 
-#include <cryptopp/osrng.h>
-using CryptoPP::AutoSeededRandomPool;
+#include "Zerocash.h"
+#include "Address.h"
 
-#include <cryptopp/eccrypto.h>
-using CryptoPP::ECP;
-using CryptoPP::ECIES;
+namespace libzerocash {
 
-#include <cryptopp/oids.h>
-namespace ASN1 = CryptoPP::ASN1;
+PrivateAddress::PrivateAddress() {
+    this->sk_enc = "";
+}
 
-#include <cryptopp/filters.h>
-using CryptoPP::StringSink;
-using CryptoPP::StringStore;
+void PrivateAddress::createPrivateAddress(const std::vector<unsigned char> a_sk, const std::string sk_enc) {
+    this->a_sk = a_sk;
+    this->sk_enc = sk_enc;
+}
 
-#include "Zerocash.h"
-#include "Address.h"
+bool PrivateAddress::operator==(const PrivateAddress& rhs) const {
```

Comment:
(Does not block this PR.) This is comparing secrets, and it's probably a security vulnerability because it's not constant time. Perhaps the easiest improvement is nuking this method, and then removing all key comparisons if possible. If that's not possible, let's do constant time comparison (which may be subtle or tricky).